### PR TITLE
crowbar_batch build: improve YAML file error handling

### PR DESCRIPTION
**Pull request by thutterer**
- abort with a better error message if the exported YAML file contains aliases that are unknown to Crowbar
- abort if the YAML file is empy (bugfix)
- replace a warn&exit two-liner with a simple abort

### CODE REVIEW

**Comment by aspiers** [comment]
+1

**Comment by thutterer** [comment]
- abort if the file does not exist

**Comment by aspiers** [comment]
Still +1

**Comment by rsalevsky** [comment]
Codewise +1

File: bin/crowbar_batch [hunk 1]
```diff
@@ -89,21 +89,27 @@ def build(input_file)
 
   host_by_alias = get_aliases[0]
 
-  input_data = File.open(input_file).readlines.join('')
+  abort "#{input_file} does not exist" unless File.file?(input_file)
```
Comment (tboerger) [inline bin/crowbar_batch]: IMHO it's more conventional to use ```File.exist?```

File: bin/crowbar_batch [hunk 2]
```diff
@@ -89,21 +89,27 @@ def build(input_file)
...
+  input_data = IO.read(input_file)
 
   # Translate @@alias@@ to Chef node name
   host_by_alias.each do |aliaz, hostname|
```
Comment (tboerger) [inline bin/crowbar_batch]: I think you can write ```alias``` as well, maybe it's just highlighted bad in your editor.

File: bin/crowbar_batch (hunk 1 above)
Comment (thutterer) [inline bin/crowbar_batch]: ```File.exist?``` would also be true with directories.

File: bin/crowbar_batch (hunk 2 above)
Comment (aspiers) [inline bin/crowbar_batch]: `alias` is a Ruby keyword ... http://ruby-doc.org/core-2.2.2/doc/keywords_rdoc.html

Comment (tboerger) [inline bin/crowbar_batch]: Really? Never heard about it :)

Comment (aspiers) [inline bin/crowbar_batch]: Yeah, people write `aliaz` and `klass` for the same reason ...

**Comment by thutterer** [comment]
@tboerger any objections to merge this as it is? it got +2
or do we want to wait for other people's review, @aspiers?

**Comment by aspiers** [comment]
I already gave +1 ... twice :-)

**Comment by thutterer** [comment]
> I already gave +1 ... twice :-)

I asked for your opinion to keep it open or merge it, not a third +1 :)

**Pull request merged**